cubic-server/command_parser/CommandParser.cpp

```diff
@@ -7,11 +7,14 @@
 #include "World.hpp"
 #include "logging/logging.hpp"
 
-void command_parser::parseCommand(std::string &command, Player *invoker)
+void command_parser::parseCommand(std::string &command, std::shared_ptr<Player> invoker)
```

Comment:
Why do you change everything to a shared_ptr for the commands? The reference counting while work for every single function call with that, and I am pretty sure that the commands don't need to own a Player, but only to reference to it, so a reference with an std::optional or a raw pointer is fine here.